libraries/AP_HAL_ChibiOS/hwdef/MambaF405/hwdef.dat

```diff
@@ -0,0 +1,145 @@
+# hw definition file for processing by chibios_pins.py
+# Mamba F405 MK2
+# with F405 mcu, mpu6000 imu and 7456 series osd
+
+# MCU class and specific type.
+MCU STM32F4xx STM32F405xx
+
+HAL_CHIBIOS_ARCH_F405 1
+
+# board ID for firmware load
+APJ_BOARD_ID 1019
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+STM32_PLLM_VALUE 8
+
+# board voltage
+STM32_VDD 330U
+
+STM32_ST_USE_TIMER 5
+
+# order of I2C buses
+I2C_ORDER I2C1
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART6 USART3 USART1
```

Comment:
This order dictates what SERIAL1/2/3 will be - have a think about whether its better to align the numbers with the documented UART numbers. You can put gaps in this list.

Reply:
If we start from OTG (USB) and go clockwise, the order of the serial ports on the board is indeed OTG1 USART6 USART3 USART1
![See https://cdn.shopify.com/s/files/1/0027/2708/4144/files/19_F405.jpg](https://cdn.shopify.com/s/files/1/0027/2708/4144/files/19_F405.jpg)

Reply:
Ok I get it, we may want to align the number of the UARTS on the board with the SERIAL number on ArduPilot
I check if there is a convention or something

Reply:
Right so currently someone will see TX6/RX6 and try and set SERIAL6 and get confused (maybe)

Reply:
All right so from https://ardupilot.org/plane/docs/parameters.html#serial-parameters I get the SERIAL order, along with how I intend to link them :
SERIAL0 => USB      => OTG1
SERIAL1 => Telem1  =>  USART6 (Still wondering if to do so knowing that on this board USART6 is tied to the ESCs connector, and usually used for ESC telemetry)
SERIAL2 => Telem2  => not present on this board
SERIAL3 => GPS       => USART3
From SERIAL4 there is no specific assignment on Ardupilot, if I am right

Reply:
and USART1 is used for the RC in so I don't know which SERIAL to use for this one

Reply:
Ok I will push this :
SERIAL0 => OTG1 (USB)
SERIAL1 => USART1 (SBUS / PPM in / TX/RX1)
SERIAL2 => EMPTY
SERIAL3 => USART3 (TX/RX3)
SERIAL4 => EMPTY
SERIAL5 => EMPTY
SERIAL6 => USART6 (ESC Telem / TX/RX6)

Reply:
Makes sense to me - I suspect anything else is going to be confusing